src/main/java/com/yelp/nrtsearch/server/luceneserver/SearchHandler.java

```diff
@@ -191,7 +191,7 @@ public SearchResponse handle(IndexState indexState, SearchRequest searchRequest)
 
       diagnostics.setFirstPassSearchTimeMs(((System.nanoTime() - searchStartTime) / 1000000.0));
 
-      DeadlineUtils.checkDeadline("SearchHandler: post recall", "SEARCH");
+      DeadlineUtils.checkDeadline("SearchHandler: post recall", diagnostics.build(), "SEARCH");
```

Comment:
@sarthakn7 I am a bit unsure about this part. Is it okay to log the build diagnostics or should I do some string conversion before logging? What would be the preferred way?

Reply:
You can convert it to json before printing if you want to, but it's toString is also fine. You can also just log the Builder object rather than calling build.

Reply:
Ok. I used toString() in this case